<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Using the BIUMmisc package for DE analyses • BIUMmisc</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using the BIUMmisc package for DE analyses">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BIUMmisc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/BIUMmisc_DE.html">Using the BIUMmisc package for DE analyses</a></li>
    <li><a class="dropdown-item" href="../articles/BIUMmisc_manual.html">Using the BIUMmisc package</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/imbeimainz/BIUMmisc/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Using the BIUMmisc package for DE analyses</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/imbeimainz/BIUMmisc/blob/main/vignettes/BIUMmisc_DE.Rmd" class="external-link"><code>vignettes/BIUMmisc_DE.Rmd</code></a></small>
      <div class="d-none name"><code>BIUMmisc_DE.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/imbeimainz/BIUMmisc" class="external-link">"BIUMmisc"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p><strong>Compiled date</strong>: 2024-07-01</p>
<p><strong>Last edited</strong>: 2024-07-01</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/thelovelab/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"topGO"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: graph</span></span>
<span><span class="co">#&gt; Loading required package: GO.db</span></span>
<span><span class="co">#&gt; Loading required package: AnnotationDbi</span></span>
<span><span class="co">#&gt; Loading required package: SparseM</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; groupGOTerms:    GOBPTerm, GOMFTerm, GOCCTerm environments built.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'topGO'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     members</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># library("pcaExplorer")</span></span>
<span><span class="co"># library("ideal")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/federicomarini/GeneTonic" class="external-link">"GeneTonic"</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="before-running-de-steps">Before running DE steps<a class="anchor" aria-label="anchor" href="#before-running-de-steps"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Load the expression data as <code>DESeqDataSet</code> object and
create the associated annotation table.<br>
Similar to Basic Protocol 3, load first the required packages, and
create the fundamental <code>DESeqDataSet</code> object to be used for
the analysis (using ENSEMBL identifiers) - optionally, one can filter
the set of lowly expressed genes as specified in the chunk below.
Generate the corresponding annotation table for
<code>dds_macrophage</code>, and store that as
<code>anno_df</code>.</li>
</ol>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Loading the packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/federicomarini/pcaExplorer" class="external-link">"pcaExplorer"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Welcome to pcaExplorer v2.31.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use pcaExplorer in your work, please cite:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Federico Marini, Harald Binder</span></span>
<span><span class="co">#&gt; pcaExplorer: an R/Bioconductor package for interacting with RNA-seq principal components</span></span>
<span><span class="co">#&gt; BMC Bioinformatics, 2019 - https://doi.org/10.1186/s12859-019-2879-1</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/federicomarini/ideal" class="external-link">"ideal"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Welcome to ideal v1.29.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use ideal in your work, please cite:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Federico Marini, Jan Linke, Harald Binder</span></span>
<span><span class="co">#&gt; ideal: an R/Bioconductor package for Interactive Differential Expression Analysis</span></span>
<span><span class="co">#&gt; BMC Bioinformatics, 2020 - https://doi.org/10.1186/s12859-020-03819-5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'ideal'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:BIUMmisc':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     sepguesser</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/federicomarini/GeneTonic" class="external-link">"GeneTonic"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loading the data</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"macrophage"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/thelovelab/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"gse"</span>, package <span class="op">=</span> <span class="st">"macrophage"</span><span class="op">)</span></span>
<span><span class="va">dds_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">gse</span>, design <span class="op">=</span> <span class="op">~</span><span class="va">line</span> <span class="op">+</span> <span class="va">condition</span><span class="op">)</span></span>
<span><span class="co">#&gt; using counts and average transcript lengths from tximeta</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="co"># one can also now use</span></span>
<span></span>
<span><span class="va">dds_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dds_gencode_to_ensembl.html">dds_gencode_to_ensembl</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Changing the ids, removing the GENCODE-specific suffix</span></span>
<span><span class="co"># rownames(dds_macrophage) &lt;- substr(rownames(dds_macrophage), 1, 15)</span></span>
<span><span class="co"># dds_macrophage</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="co"># Filtering low expressed features</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">6</span> </span>
<span><span class="va">dds_macrophage</span> <span class="op">&lt;-</span> <span class="va">dds_macrophage</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span></span>
<span><span class="va">dds_macrophage</span></span>
<span><span class="co">#&gt; class: DESeqDataSet </span></span>
<span><span class="co">#&gt; dim: 17806 24 </span></span>
<span><span class="co">#&gt; metadata(7): tximetaInfo quantInfo ... txdbInfo version</span></span>
<span><span class="co">#&gt; assays(3): counts abundance avgTxLength</span></span>
<span><span class="co">#&gt; rownames(17806): ENSG00000000003 ENSG00000000419 ... ENSG00000285982</span></span>
<span><span class="co">#&gt;   ENSG00000285994</span></span>
<span><span class="co">#&gt; rowData names(3): gene_id SYMBOL gencode_id</span></span>
<span><span class="co">#&gt; colnames(24): SAMEA103885102 SAMEA103885347 ... SAMEA103885308</span></span>
<span><span class="co">#&gt;   SAMEA103884949</span></span>
<span><span class="co">#&gt; colData names(15): names sample_id ... condition line</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Construct the annotation data frame</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="va">anno_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://federicomarini.github.io/pcaExplorer/reference/get_annotation_orgdb.html" class="external-link">get_annotation_orgdb</a></span><span class="op">(</span>dds <span class="op">=</span> <span class="va">dds_macrophage</span>,</span>
<span>                                orgdb_species <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span>                                idtype <span class="op">=</span> <span class="st">"ENSEMBL"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/grimbough/biomaRt" class="external-link">"biomaRt"</a></span><span class="op">)</span></span>
<span><span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span>biomart<span class="op">=</span><span class="st">"ENSEMBL_MART_ENSEMBL"</span>, dataset<span class="op">=</span><span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">anns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html" class="external-link">getBM</a></span><span class="op">(</span>attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span>, <span class="st">"description"</span><span class="op">)</span>, </span>
<span>              filters <span class="op">=</span> <span class="st">"ensembl_gene_id"</span>,</span>
<span>              <span class="co"># filters = "external_gene_name",</span></span>
<span>              values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span>, </span>
<span>              mart <span class="op">=</span> <span class="va">mart</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_totcounts.html">plot_totcounts</a></span><span class="op">(</span><span class="va">dds_macrophage</span>, group <span class="op">=</span> <span class="st">"condition"</span><span class="op">)</span></span></code></pre></div>
<p><img src="BIUMmisc_DE_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">extended_anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fortify_annotations.html">fortify_annotations</a></span><span class="op">(</span>anno_df_1 <span class="op">=</span> <span class="va">anno_df</span>,</span>
<span>                                     anno_df_2 <span class="op">=</span> <span class="va">anns</span>,</span>
<span>                                     dds <span class="op">=</span> <span class="va">dds_macrophage</span><span class="op">)</span></span>
<span><span class="co">#&gt; Found 2288 features with value NA in column gene_name (anno_df_1)</span></span>
<span><span class="co">#&gt; Found 137 features with value NA in column external_gene_name (anno_df_2)</span></span>
<span><span class="co">#&gt; Trying to resolve missing info in gene_name (anno_df_1) with info from external_gene_name (anno_df_2)...</span></span>
<span><span class="co">#&gt; Found 127 features with value NA in the resolved gene name column</span></span>
<span><span class="co">#&gt; You might want to add/edit additionally this annotation table...</span></span>
<span><span class="co">#&gt; Updated rowData slot of the dds object by adding columns: anno_gene_id, anno_gene_name, anno_description</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">extended_anno</span><span class="op">$</span><span class="va">anno_df</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         gene_id gene_name external_gene_name</span></span>
<span><span class="co">#&gt; ENSG00000000003 ENSG00000000003    TSPAN6             TSPAN6</span></span>
<span><span class="co">#&gt; ENSG00000000419 ENSG00000000419      DPM1               DPM1</span></span>
<span><span class="co">#&gt; ENSG00000000457 ENSG00000000457     SCYL3              SCYL3</span></span>
<span><span class="co">#&gt; ENSG00000000460 ENSG00000000460     FIRRM              FIRRM</span></span>
<span><span class="co">#&gt; ENSG00000000938 ENSG00000000938       FGR                FGR</span></span>
<span><span class="co">#&gt; ENSG00000000971 ENSG00000000971       CFH                CFH</span></span>
<span><span class="co">#&gt;                                                                                                    description</span></span>
<span><span class="co">#&gt; ENSG00000000003                                              tetraspanin 6 [Source:HGNC Symbol;Acc:HGNC:11858]</span></span>
<span><span class="co">#&gt; ENSG00000000419 dolichyl-phosphate mannosyltransferase subunit 1, catalytic [Source:HGNC Symbol;Acc:HGNC:3005]</span></span>
<span><span class="co">#&gt; ENSG00000000457                                   SCY1 like pseudokinase 3 [Source:HGNC Symbol;Acc:HGNC:19285]</span></span>
<span><span class="co">#&gt; ENSG00000000460  FIGNL1 interacting regulator of recombination and mitosis [Source:HGNC Symbol;Acc:HGNC:25565]</span></span>
<span><span class="co">#&gt; ENSG00000000938              FGR proto-oncogene, Src family tyrosine kinase [Source:HGNC Symbol;Acc:HGNC:3697]</span></span>
<span><span class="co">#&gt; ENSG00000000971                                         complement factor H [Source:HGNC Symbol;Acc:HGNC:4883]</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">extended_anno</span><span class="op">$</span><span class="va">dds</span></span>
<span><span class="co">#&gt; class: DESeqDataSet </span></span>
<span><span class="co">#&gt; dim: 17806 24 </span></span>
<span><span class="co">#&gt; metadata(7): tximetaInfo quantInfo ... txdbInfo version</span></span>
<span><span class="co">#&gt; assays(3): counts abundance avgTxLength</span></span>
<span><span class="co">#&gt; rownames(17806): ENSG00000000003 ENSG00000000419 ... ENSG00000285982</span></span>
<span><span class="co">#&gt;   ENSG00000285994</span></span>
<span><span class="co">#&gt; rowData names(6): gene_id SYMBOL ... anno_gene_name anno_description</span></span>
<span><span class="co">#&gt; colnames(24): SAMEA103885102 SAMEA103885347 ... SAMEA103885308</span></span>
<span><span class="co">#&gt;   SAMEA103884949</span></span>
<span><span class="co">#&gt; colData names(15): names sample_id ... condition line</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_normcounts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deseq_normcounts_with_info.html">deseq_normcounts_with_info</a></span><span class="op">(</span>dds <span class="op">=</span> <span class="va">dds_macrophage</span>,</span>
<span>                                             extended_anno_df <span class="op">=</span> <span class="va">extended_anno</span><span class="op">$</span><span class="va">anno_df</span><span class="op">)</span></span>
<span><span class="co">#&gt; using 'avgTxLength' from assays(dds), correcting for library size</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">tbl_TPMs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deseq_tpm_with_info.html">deseq_tpm_with_info</a></span><span class="op">(</span>dds <span class="op">=</span> <span class="va">dds_macrophage</span>,</span>
<span>                                extended_anno_df <span class="op">=</span> <span class="va">extended_anno</span><span class="op">$</span><span class="va">anno_df</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="running-the-de-steps">Running the DE steps<a class="anchor" aria-label="anchor" href="#running-the-de-steps"></a>
</h2>
<p>Creating the DE results themselves…</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds_macrophage</span></span>
<span><span class="co">#&gt; class: DESeqDataSet </span></span>
<span><span class="co">#&gt; dim: 17806 24 </span></span>
<span><span class="co">#&gt; metadata(7): tximetaInfo quantInfo ... txdbInfo version</span></span>
<span><span class="co">#&gt; assays(3): counts abundance avgTxLength</span></span>
<span><span class="co">#&gt; rownames(17806): ENSG00000000003 ENSG00000000419 ... ENSG00000285982</span></span>
<span><span class="co">#&gt;   ENSG00000285994</span></span>
<span><span class="co">#&gt; rowData names(3): gene_id SYMBOL gencode_id</span></span>
<span><span class="co">#&gt; colnames(24): SAMEA103885102 SAMEA103885347 ... SAMEA103885308</span></span>
<span><span class="co">#&gt;   SAMEA103884949</span></span>
<span><span class="co">#&gt; colData names(15): names sample_id ... condition line</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">design</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span>
<span><span class="co">#&gt; ~line + condition</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span>
<span><span class="co">#&gt; estimating size factors</span></span>
<span><span class="co">#&gt; using 'avgTxLength' from assays(dds), correcting for library size</span></span>
<span><span class="co">#&gt; estimating dispersions</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; fitting model and testing</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">resultsNames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Intercept"                      "line_eiwy_1_vs_diku_1"         </span></span>
<span><span class="co">#&gt; [3] "line_fikt_3_vs_diku_1"          "line_ieki_2_vs_diku_1"         </span></span>
<span><span class="co">#&gt; [5] "line_podx_1_vs_diku_1"          "line_qaqx_1_vs_diku_1"         </span></span>
<span><span class="co">#&gt; [7] "condition_IFNg_vs_naive"        "condition_IFNg_SL1344_vs_naive"</span></span>
<span><span class="co">#&gt; [9] "condition_SL1344_vs_naive"</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">myresuSet_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">myresuSet_macrophage</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/create_DEresults.html">create_DEresults</a></span><span class="op">(</span>resuSet <span class="op">=</span> <span class="va">myresuSet_macrophage</span>,</span>
<span>                   dds_obj <span class="op">=</span> <span class="va">dds_macrophage</span>,</span>
<span>                   contrast_name <span class="op">=</span> <span class="st">"condition_IFNg_vs_naive"</span>, </span>
<span>                   FDR <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                   extended_anno_df <span class="op">=</span> <span class="va">extended_anno</span><span class="op">$</span><span class="va">anno_df</span>,</span>
<span>                   species <span class="op">=</span> <span class="st">"Homo_sapiens"</span></span>
<span>                   <span class="op">)</span></span>
<span><span class="co">#&gt; Storing set of results in the element `condition_IFNg_vs_naive`...</span></span>
<span><span class="co">#&gt; Extracting results...</span></span>
<span><span class="co">#&gt; Performing LFC shrinkage...</span></span>
<span><span class="co">#&gt; using 'apeglm' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">#&gt;     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span></span>
<span><span class="co">#&gt;     sequence count data: removing the noise and preserving large differences.</span></span>
<span><span class="co">#&gt;     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</span></span>
<span><span class="co">#&gt; Summary MAplot...</span></span>
<span><span class="co">#&gt; Extracting tables...</span></span>
<span><span class="co">#&gt; Extracting DEtables...</span></span>
<span><span class="co">#&gt; Generating interactive DEtable...</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">myresuSet_macrophage</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/create_DEresults.html">create_DEresults</a></span><span class="op">(</span>resuSet <span class="op">=</span> <span class="va">myresuSet_macrophage</span>,</span>
<span>                   dds_obj <span class="op">=</span> <span class="va">dds_macrophage</span>,</span>
<span>                   contrast_name <span class="op">=</span> <span class="st">"condition_IFNg_vs_naive"</span>, </span>
<span>                   lfc_threshold <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>                   FDR <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                   extended_anno_df <span class="op">=</span> <span class="va">extended_anno</span><span class="op">$</span><span class="va">anno_df</span>,</span>
<span>                   species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>,</span>
<span>                   name_res_entry <span class="op">=</span> <span class="st">"ifng-naive-lfc0.6"</span></span>
<span>                   <span class="op">)</span></span>
<span><span class="co">#&gt; Storing set of results in the element `ifng-naive-lfc0.6`...</span></span>
<span><span class="co">#&gt; Extracting results...</span></span>
<span><span class="co">#&gt; Performing LFC shrinkage...</span></span>
<span><span class="co">#&gt; using 'apeglm' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">#&gt;     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span></span>
<span><span class="co">#&gt;     sequence count data: removing the noise and preserving large differences.</span></span>
<span><span class="co">#&gt;     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</span></span>
<span><span class="co">#&gt; Summary MAplot...</span></span>
<span><span class="co">#&gt; Extracting tables...</span></span>
<span><span class="co">#&gt; Extracting DEtables...</span></span>
<span><span class="co">#&gt; Generating interactive DEtable...</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">myresuSet_macrophage</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/create_DEresults.html">create_DEresults</a></span><span class="op">(</span>resuSet <span class="op">=</span> <span class="va">myresuSet_macrophage</span>,</span>
<span>                   dds_obj <span class="op">=</span> <span class="va">dds_macrophage</span>,</span>
<span>                   contrast_name <span class="op">=</span> <span class="st">"condition_IFNg_vs_naive"</span>, </span>
<span>                   lfc_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                   FDR <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                   extended_anno_df <span class="op">=</span> <span class="va">extended_anno</span><span class="op">$</span><span class="va">anno_df</span>,</span>
<span>                   species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>,</span>
<span>                   name_res_entry <span class="op">=</span> <span class="st">"ifng-naive-lfc1"</span></span>
<span>                   <span class="op">)</span></span>
<span><span class="co">#&gt; Storing set of results in the element `ifng-naive-lfc1`...</span></span>
<span><span class="co">#&gt; Extracting results...</span></span>
<span><span class="co">#&gt; Performing LFC shrinkage...</span></span>
<span><span class="co">#&gt; using 'apeglm' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">#&gt;     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span></span>
<span><span class="co">#&gt;     sequence count data: removing the noise and preserving large differences.</span></span>
<span><span class="co">#&gt;     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</span></span>
<span><span class="co">#&gt; Summary MAplot...</span></span>
<span><span class="co">#&gt; Extracting tables...</span></span>
<span><span class="co">#&gt; Extracting DEtables...</span></span>
<span><span class="co">#&gt; Generating interactive DEtable...</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"ENSG00000125347"</span>,   <span class="co"># IRF1</span></span>
<span>  <span class="st">"ENSG00000110944"</span>,   <span class="co"># IL23A</span></span>
<span>  <span class="st">"ENSG00000084636"</span>,   <span class="co"># COL16A1</span></span>
<span>  <span class="st">"ENSG00000172399"</span>    <span class="co"># MYOZ2</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://federicomarini.github.io/ideal/reference/plot_ma.html" class="external-link">plot_ma</a></span><span class="op">(</span>res_obj <span class="op">=</span> <span class="va">myresuSet_macrophage</span><span class="op">$</span><span class="va">`ifng-naive-lfc1`</span><span class="op">$</span><span class="va">res_DESeq</span>, </span>
<span>        intgenes <span class="op">=</span> <span class="va">genes_of_interest</span><span class="op">)</span></span></code></pre></div>
<p><img src="BIUMmisc_DE_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://federicomarini.github.io/ideal/reference/plot_volcano.html" class="external-link">plot_volcano</a></span><span class="op">(</span>res_obj <span class="op">=</span> <span class="va">myresuSet_macrophage</span><span class="op">$</span><span class="va">`ifng-naive-lfc1`</span><span class="op">$</span><span class="va">res_DESeq</span>,</span>
<span>             ylim_up <span class="op">=</span> <span class="fl">40</span>,</span>
<span>             intgenes <span class="op">=</span> <span class="va">genes_of_interest</span><span class="op">)</span></span></code></pre></div>
<p><img src="BIUMmisc_DE_files/figure-html/unnamed-chunk-9-2.png" width="700"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># TODO: adjust the y limit</span></span>
<span><span class="co"># TODO: add text ready to be ggplotlified?</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="after-the-de-steps-functional-enrichment">After the DE steps: functional enrichment<a class="anchor" aria-label="anchor" href="#after-the-de-steps-functional-enrichment"></a>
</h2>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">extended_anno_df</span> <span class="op">&lt;-</span> <span class="va">extended_anno</span><span class="op">$</span><span class="va">anno_df</span></span>
<span><span class="va">expressedInAssay</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">geneUniverseExprENS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span><span class="op">[</span><span class="va">expressedInAssay</span><span class="op">]</span></span>
<span><span class="va">geneUniverseExpr</span> <span class="op">&lt;-</span> <span class="va">extended_anno_df</span><span class="op">$</span><span class="va">gene_name</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">geneUniverseExprENS</span>, <span class="va">extended_anno_df</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>We iterate on all contrasts in the <code>myresuSet</code> object</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">myresuSet_macrophage</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">myresuSet_macrophage</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"tbl_res_DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">myresuSet_macrophage</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"topGO_tbl"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://federicomarini.github.io/pcaExplorer/reference/topGOtable.html" class="external-link">topGOtable</a></span><span class="op">(</span>DEgenes <span class="op">=</span> <span class="va">myresuSet_macrophage</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"tbl_res_DE"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">gene_name</span>, </span>
<span>               BGgenes <span class="op">=</span> <span class="va">geneUniverseExpr</span>, </span>
<span>               ontology <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>               geneID <span class="op">=</span> <span class="st">"symbol"</span>,</span>
<span>               addGeneToTerms<span class="op">=</span><span class="cn">TRUE</span>, </span>
<span>               topTablerows <span class="op">=</span> <span class="fl">500</span>,</span>
<span>               mapping <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; condition_IFNg_vs_naive</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Building most specific GOs .....</span></span>
<span><span class="co">#&gt;  ( 11445 GO terms found. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Build GO DAG topology ..........</span></span>
<span><span class="co">#&gt;  ( 14788 GO terms and 33231 relations. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Annotating nodes ...............</span></span>
<span><span class="co">#&gt;  ( 13154 genes annotated to the GO terms. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           -- Elim Algorithm -- </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       the algorithm is scoring 6114 nontrivial nodes</span></span>
<span><span class="co">#&gt;       parameters: </span></span>
<span><span class="co">#&gt;           test statistic: fisher</span></span>
<span><span class="co">#&gt;           cutOff: 0.01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 19:  2 nodes to be scored    (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 18:  3 nodes to be scored    (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 17:  9 nodes to be scored    (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 16:  32 nodes to be scored   (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 15:  49 nodes to be scored   (117 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 14:  66 nodes to be scored   (172 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 13:  118 nodes to be scored  (293 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 12:  270 nodes to be scored  (451 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 11:  491 nodes to be scored  (754 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 10:  730 nodes to be scored  (1400 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 9:   871 nodes to be scored  (1886 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 8:   933 nodes to be scored  (2732 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 7:   917 nodes to be scored  (3937 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 6:   779 nodes to be scored  (4746 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 5:   468 nodes to be scored  (5476 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 4:   262 nodes to be scored  (5732 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 3:   96 nodes to be scored   (6108 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 2:   17 nodes to be scored   (8342 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 1:   1 nodes to be scored    (8342 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           -- Classic Algorithm -- </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       the algorithm is scoring 6114 nontrivial nodes</span></span>
<span><span class="co">#&gt;       parameters: </span></span>
<span><span class="co">#&gt;           test statistic: fisher</span></span>
<span><span class="co">#&gt; Warning in topGOtable(DEgenes =</span></span>
<span><span class="co">#&gt; myresuSet_macrophage[[i]][["tbl_res_DE"]]$gene_name, : NAs introduced by</span></span>
<span><span class="co">#&gt; coercion</span></span>
<span><span class="co">#&gt; ifng-naive-lfc0.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Building most specific GOs .....</span></span>
<span><span class="co">#&gt;  ( 11445 GO terms found. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Build GO DAG topology ..........</span></span>
<span><span class="co">#&gt;  ( 14788 GO terms and 33231 relations. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Annotating nodes ...............</span></span>
<span><span class="co">#&gt;  ( 13154 genes annotated to the GO terms. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           -- Elim Algorithm -- </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       the algorithm is scoring 5607 nontrivial nodes</span></span>
<span><span class="co">#&gt;       parameters: </span></span>
<span><span class="co">#&gt;           test statistic: fisher</span></span>
<span><span class="co">#&gt;           cutOff: 0.01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 19:  2 nodes to be scored    (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 18:  3 nodes to be scored    (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 17:  8 nodes to be scored    (12 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 16:  29 nodes to be scored   (19 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 15:  44 nodes to be scored   (118 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 14:  58 nodes to be scored   (220 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 13:  100 nodes to be scored  (326 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 12:  228 nodes to be scored  (400 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 11:  421 nodes to be scored  (709 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 10:  655 nodes to be scored  (1387 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 9:   775 nodes to be scored  (2159 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 8:   857 nodes to be scored  (2811 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 7:   864 nodes to be scored  (3608 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 6:   749 nodes to be scored  (4700 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 5:   450 nodes to be scored  (5276 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 4:   253 nodes to be scored  (6090 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 3:   93 nodes to be scored   (7118 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 2:   17 nodes to be scored   (7584 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 1:   1 nodes to be scored    (7584 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           -- Classic Algorithm -- </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       the algorithm is scoring 5607 nontrivial nodes</span></span>
<span><span class="co">#&gt;       parameters: </span></span>
<span><span class="co">#&gt;           test statistic: fisher</span></span>
<span><span class="co">#&gt; Warning in topGOtable(DEgenes =</span></span>
<span><span class="co">#&gt; myresuSet_macrophage[[i]][["tbl_res_DE"]]$gene_name, : NAs introduced by</span></span>
<span><span class="co">#&gt; coercion</span></span>
<span><span class="co">#&gt; ifng-naive-lfc1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Building most specific GOs .....</span></span>
<span><span class="co">#&gt;  ( 11445 GO terms found. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Build GO DAG topology ..........</span></span>
<span><span class="co">#&gt;  ( 14788 GO terms and 33231 relations. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Annotating nodes ...............</span></span>
<span><span class="co">#&gt;  ( 13154 genes annotated to the GO terms. )</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           -- Elim Algorithm -- </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       the algorithm is scoring 5010 nontrivial nodes</span></span>
<span><span class="co">#&gt;       parameters: </span></span>
<span><span class="co">#&gt;           test statistic: fisher</span></span>
<span><span class="co">#&gt;           cutOff: 0.01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 19:  2 nodes to be scored    (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 18:  3 nodes to be scored    (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 17:  7 nodes to be scored    (0 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 16:  25 nodes to be scored   (26 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 15:  40 nodes to be scored   (57 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 14:  46 nodes to be scored   (87 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 13:  75 nodes to be scored   (244 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 12:  194 nodes to be scored  (310 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 11:  349 nodes to be scored  (574 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 10:  561 nodes to be scored  (972 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 9:   682 nodes to be scored  (1505 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 8:   768 nodes to be scored  (2174 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 7:   791 nodes to be scored  (2817 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 6:   690 nodes to be scored  (3303 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 5:   427 nodes to be scored  (3956 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 4:   241 nodes to be scored  (4122 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 3:   91 nodes to be scored   (4619 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 2:   17 nodes to be scored   (5149 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Level 1:   1 nodes to be scored    (5149 eliminated genes)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           -- Classic Algorithm -- </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       the algorithm is scoring 5010 nontrivial nodes</span></span>
<span><span class="co">#&gt;       parameters: </span></span>
<span><span class="co">#&gt;           test statistic: fisher</span></span>
<span><span class="co">#&gt; Warning in topGOtable(DEgenes =</span></span>
<span><span class="co">#&gt; myresuSet_macrophage[[i]][["tbl_res_DE"]]$gene_name, : NAs introduced by</span></span>
<span><span class="co">#&gt; coercion</span></span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">"clusterProfiler"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; clusterProfiler v4.13.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use clusterProfiler in published research, please cite:</span></span>
<span><span class="co">#&gt; T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'clusterProfiler'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:biomaRt':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     select</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:AnnotationDbi':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     select</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     slice</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:S4Vectors':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rename</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">myresuSet_macrophage</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">myresuSet_macrophage</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"tbl_res_DE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">myresuSet_macrophage</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"clupro_tbl"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>      <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span></span>
<span>        gene <span class="op">=</span> <span class="va">myresuSet_macrophage</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"tbl_res_DE"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">gene_name</span>,</span>
<span>        universe      <span class="op">=</span> <span class="va">geneUniverseExpr</span>,</span>
<span>        keyType       <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>        OrgDb         <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,</span>
<span>        ont           <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>        pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>        pvalueCutoff  <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>        qvalueCutoff  <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        readable      <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; condition_IFNg_vs_naive</span></span>
<span><span class="co">#&gt; ifng-naive-lfc0.6</span></span>
<span><span class="co">#&gt; ifng-naive-lfc1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="wrapping-up-a-de-report">Wrapping up a DE report<a class="anchor" aria-label="anchor" href="#wrapping-up-a-de-report"></a>
</h2>
<p>transforming this into a gtl</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gtl_macs_ifng_naive</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resuset_to_gtl.html">resuset_to_gtl</a></span><span class="op">(</span><span class="va">myresuSet_macrophage</span>, </span>
<span>                                      result_name <span class="op">=</span> <span class="st">"ifng-naive-lfc0.6"</span>,</span>
<span>                                      dds <span class="op">=</span> <span class="va">dds_macrophage</span>,</span>
<span>                                      anno_df <span class="op">=</span> <span class="va">extended_anno_df</span><span class="op">)</span></span>
<span><span class="co">#&gt; Found 500 gene sets in `topGOtableResult` object.</span></span>
<span><span class="co">#&gt; Converting for usage in GeneTonic...</span></span>
<span><span class="co">#&gt; ---------------------------------</span></span>
<span><span class="co">#&gt; ----- GeneTonicList object ------</span></span>
<span><span class="co">#&gt; ---------------------------------</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ----- dds object -----</span></span>
<span><span class="co">#&gt; Providing an expression object (as DESeqDataset) of 17806 features over 24 samples</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ----- res_de object -----</span></span>
<span><span class="co">#&gt; Providing a DE result object (as DESeqResults), 17806 features tested, 1870 found as DE</span></span>
<span><span class="co">#&gt; Upregulated:     1109</span></span>
<span><span class="co">#&gt; Downregulated:   761</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ----- res_enrich object -----</span></span>
<span><span class="co">#&gt; Providing an enrichment result object, 500 reported</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ----- annotation_obj object -----</span></span>
<span><span class="co">#&gt; Providing an annotation object of 17806 features with information on 4 identifier types</span></span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GeneTonic/man/describe_gtl.html" class="external-link">describe_gtl</a></span><span class="op">(</span><span class="va">gtl_macs_ifng_naive</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "---------------------------------\n"                                                      </span></span>
<span><span class="co">#&gt;  [2] "----- GeneTonicList object ------\n"                                                      </span></span>
<span><span class="co">#&gt;  [3] "---------------------------------\n"                                                      </span></span>
<span><span class="co">#&gt;  [4] "\n----- dds object -----\n"                                                               </span></span>
<span><span class="co">#&gt;  [5] "Providing an expression object (as DESeqDataset) of 17806 features over 24 samples\n"     </span></span>
<span><span class="co">#&gt;  [6] "\n----- res_de object -----\n"                                                            </span></span>
<span><span class="co">#&gt;  [7] "Providing a DE result object (as DESeqResults), 17806 features tested, 1870 found as DE\n"</span></span>
<span><span class="co">#&gt;  [8] "Upregulated:     1109\n"                                                                  </span></span>
<span><span class="co">#&gt;  [9] "Downregulated:   761\n"                                                                   </span></span>
<span><span class="co">#&gt; [10] "\n----- res_enrich object -----\n"                                                        </span></span>
<span><span class="co">#&gt; [11] "Providing an enrichment result object, 500 reported\n"                                    </span></span>
<span><span class="co">#&gt; [12] "\n----- annotation_obj object -----\n"                                                    </span></span>
<span><span class="co">#&gt; [13] "Providing an annotation object of 17806 features with information on 4 identifier types\n"</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">vst_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/vst.html" class="external-link">vst</a></span><span class="op">(</span><span class="va">dds_macrophage</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GeneTonic/man/gs_heatmap.html" class="external-link">gs_heatmap</a></span><span class="op">(</span>se <span class="op">=</span> <span class="va">vst_macrophage</span>,</span>
<span>           gtl <span class="op">=</span> <span class="va">gtl_macs_ifng_naive</span>,</span>
<span>           geneset_id <span class="op">=</span> <span class="st">"GO:0034341"</span>,</span>
<span>           cluster_columns <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           anno_col_info <span class="op">=</span> <span class="st">"condition"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BIUMmisc_DE_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GeneTonic/man/signature_volcano.html" class="external-link">signature_volcano</a></span><span class="op">(</span>gtl <span class="op">=</span> <span class="va">gtl_macs_ifng_naive</span>,</span>
<span>                  geneset_id <span class="op">=</span> <span class="st">"GO:0034341"</span>,</span>
<span>                  FDR <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 23 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="BIUMmisc_DE_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GeneTonic/man/gs_scoresheat.html" class="external-link">gs_scoresheat</a></span><span class="op">(</span></span>
<span>  mat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GeneTonic/man/gs_scores.html" class="external-link">gs_scores</a></span><span class="op">(</span></span>
<span>    se <span class="op">=</span> <span class="va">vst_macrophage</span>,</span>
<span>    gtl <span class="op">=</span> <span class="va">gtl_macs_ifng_naive</span><span class="op">)</span>,</span>
<span>  n_gs <span class="op">=</span> <span class="fl">30</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BIUMmisc_DE_files/figure-html/unnamed-chunk-11-3.png" width="700"></p>
<p>exporting them all</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/exportr.html">exportr</a></span><span class="op">(</span><span class="va">myresuSet_macrophage</span>,</span>
<span>        out_file_prefix <span class="op">=</span> <span class="st">"fedetest"</span>, out_folder <span class="op">=</span> <span class="st">"dirtest"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Exporting the whole set for fedetest</span></span>
<span><span class="co">#&gt;     fedetest  ---  condition_IFNg_vs_naive</span></span>
<span><span class="co">#&gt;         Exporting results from DESeq...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_condition_IFNg_vs_naive_tbl_res_DESeq.xlsx</span></span>
<span><span class="co">#&gt;         Exporting results from topGO...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_condition_IFNg_vs_naive_tbl_topGOres.xlsx</span></span>
<span><span class="co">#&gt;         Exporting results from clusterProfiler...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_condition_IFNg_vs_naive_tbl_cluPro.xlsx</span></span>
<span><span class="co">#&gt;     fedetest  ---  ifng-naive-lfc0.6</span></span>
<span><span class="co">#&gt;         Exporting results from DESeq...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_ifng-naive-lfc0.6_tbl_res_DESeq.xlsx</span></span>
<span><span class="co">#&gt;         Exporting results from topGO...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_ifng-naive-lfc0.6_tbl_topGOres.xlsx</span></span>
<span><span class="co">#&gt;         Exporting results from clusterProfiler...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_ifng-naive-lfc0.6_tbl_cluPro.xlsx</span></span>
<span><span class="co">#&gt;     fedetest  ---  ifng-naive-lfc1</span></span>
<span><span class="co">#&gt;         Exporting results from DESeq...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_ifng-naive-lfc1_tbl_res_DESeq.xlsx</span></span>
<span><span class="co">#&gt;         Exporting results from topGO...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_ifng-naive-lfc1_tbl_topGOres.xlsx</span></span>
<span><span class="co">#&gt;         Exporting results from clusterProfiler...</span></span>
<span><span class="co">#&gt;           dirtest/fedetest_ifng-naive-lfc1_tbl_cluPro.xlsx</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># BiocManager::version()</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] clusterProfiler_4.13.0      biomaRt_2.61.2             </span></span>
<span><span class="co">#&gt;  [3] macrophage_1.21.0           ideal_1.29.0               </span></span>
<span><span class="co">#&gt;  [5] pcaExplorer_2.31.0          GeneTonic_2.9.0            </span></span>
<span><span class="co">#&gt;  [7] org.Hs.eg.db_3.19.1         topGO_2.57.0               </span></span>
<span><span class="co">#&gt;  [9] SparseM_1.84                GO.db_3.19.1               </span></span>
<span><span class="co">#&gt; [11] AnnotationDbi_1.67.0        graph_1.83.0               </span></span>
<span><span class="co">#&gt; [13] DESeq2_1.45.0               SummarizedExperiment_1.35.1</span></span>
<span><span class="co">#&gt; [15] Biobase_2.65.0              MatrixGenerics_1.17.0      </span></span>
<span><span class="co">#&gt; [17] matrixStats_1.3.0           GenomicRanges_1.57.1       </span></span>
<span><span class="co">#&gt; [19] GenomeInfoDb_1.41.1         IRanges_2.39.0             </span></span>
<span><span class="co">#&gt; [21] S4Vectors_0.43.0            BiocGenerics_0.51.0        </span></span>
<span><span class="co">#&gt; [23] BIUMmisc_0.1.0             </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] GSEABase_1.67.0          progress_1.2.3           DT_0.33                 </span></span>
<span><span class="co">#&gt;   [4] Biostrings_2.73.1        vctrs_0.6.5              digest_0.6.36           </span></span>
<span><span class="co">#&gt;   [7] png_0.1-8                shape_1.4.6.1            shinyBS_0.61.1          </span></span>
<span><span class="co">#&gt;  [10] registry_0.5-1           ggrepel_0.9.5            MASS_7.3-61             </span></span>
<span><span class="co">#&gt;  [13] pkgdown_2.0.9.9000       reshape2_1.4.4           httpuv_1.6.15           </span></span>
<span><span class="co">#&gt;  [16] foreach_1.5.2            qvalue_2.37.0            withr_3.0.0             </span></span>
<span><span class="co">#&gt;  [19] xfun_0.45                ggfun_0.1.5              survival_3.7-0          </span></span>
<span><span class="co">#&gt;  [22] memoise_2.0.1            gson_0.1.0               BiasedUrn_2.0.12        </span></span>
<span><span class="co">#&gt;  [25] systemfonts_1.1.0        ragg_1.3.2               tidytree_0.4.6          </span></span>
<span><span class="co">#&gt;  [28] GlobalOptions_0.1.2      gtools_3.9.5             prettyunits_1.2.0       </span></span>
<span><span class="co">#&gt;  [31] KEGGREST_1.45.1          promises_1.3.0           httr_1.4.7              </span></span>
<span><span class="co">#&gt;  [34] restfulr_0.0.15          rstudioapi_0.16.0        shinyAce_0.4.2          </span></span>
<span><span class="co">#&gt;  [37] UCSC.utils_1.1.0         miniUI_0.1.1.1           generics_0.1.3          </span></span>
<span><span class="co">#&gt;  [40] DOSE_3.31.2              base64enc_0.1-3          curl_5.2.1              </span></span>
<span><span class="co">#&gt;  [43] zlibbioc_1.51.1          ggraph_2.2.1             polyclip_1.10-6         </span></span>
<span><span class="co">#&gt;  [46] ca_0.71.1                GenomeInfoDbData_1.2.12  SparseArray_1.5.10      </span></span>
<span><span class="co">#&gt;  [49] RBGL_1.81.0              threejs_0.3.3            xtable_1.8-4            </span></span>
<span><span class="co">#&gt;  [52] stringr_1.5.1            desc_1.4.3               doParallel_1.0.17       </span></span>
<span><span class="co">#&gt;  [55] evaluate_0.24.0          S4Arrays_1.5.1           BiocFileCache_2.13.0    </span></span>
<span><span class="co">#&gt;  [58] hms_1.1.3                colorspace_2.1-0         filelock_1.0.3          </span></span>
<span><span class="co">#&gt;  [61] visNetwork_2.1.2         shinyWidgets_0.8.6       magrittr_2.0.3          </span></span>
<span><span class="co">#&gt;  [64] Rgraphviz_2.49.0         later_1.3.2              viridis_0.6.5           </span></span>
<span><span class="co">#&gt;  [67] ggtree_3.13.0            lattice_0.22-6           NMF_0.27                </span></span>
<span><span class="co">#&gt;  [70] genefilter_1.87.0        XML_3.99-0.17            shadowtext_0.1.3        </span></span>
<span><span class="co">#&gt;  [73] cowplot_1.1.3            pillar_1.9.0             nlme_3.1-165            </span></span>
<span><span class="co">#&gt;  [76] iterators_1.0.14         gridBase_0.4-7           caTools_1.18.2          </span></span>
<span><span class="co">#&gt;  [79] compiler_4.4.1           stringi_1.8.4            shinycssloaders_1.0.0   </span></span>
<span><span class="co">#&gt;  [82] Category_2.71.0          TSP_1.2-4                dendextend_1.17.1       </span></span>
<span><span class="co">#&gt;  [85] GenomicAlignments_1.41.0 plyr_1.8.9               crayon_1.5.3            </span></span>
<span><span class="co">#&gt;  [88] abind_1.4-5              BiocIO_1.15.0            gridGraphics_0.5-1      </span></span>
<span><span class="co">#&gt;  [91] emdbook_1.3.13           locfit_1.5-9.10          graphlayouts_1.1.1      </span></span>
<span><span class="co">#&gt;  [94] bit_4.0.5                UpSetR_1.4.0             dplyr_1.1.4             </span></span>
<span><span class="co">#&gt;  [97] fastmatch_1.1-4          codetools_0.2-20         textshaping_0.4.0       </span></span>
<span><span class="co">#&gt; [100] crosstalk_1.2.1          bslib_0.7.0              slam_0.1-50             </span></span>
<span><span class="co">#&gt; [103] GetoptLong_1.0.5         plotly_4.10.4            mime_0.12               </span></span>
<span><span class="co">#&gt; [106] mosdef_1.1.0             splines_4.4.1            circlize_0.4.16         </span></span>
<span><span class="co">#&gt; [109] Rcpp_1.0.12              dbplyr_2.5.0             tippy_0.1.0             </span></span>
<span><span class="co">#&gt; [112] HDO.db_0.99.1            knitr_1.47               blob_1.2.4              </span></span>
<span><span class="co">#&gt; [115] utf8_1.2.4               clue_0.3-65              apeglm_1.27.1           </span></span>
<span><span class="co">#&gt; [118] fs_1.6.4                 backbone_2.1.4           IHW_1.33.0              </span></span>
<span><span class="co">#&gt; [121] expm_0.999-9             ggplotify_0.1.2          tibble_3.2.1            </span></span>
<span><span class="co">#&gt; [124] Matrix_1.7-0             statmod_1.5.0            tweenr_2.0.3            </span></span>
<span><span class="co">#&gt; [127] pkgconfig_2.0.3          pheatmap_1.0.12          tools_4.4.1             </span></span>
<span><span class="co">#&gt; [130] cachem_1.1.0             RSQLite_2.3.7            viridisLite_0.4.2       </span></span>
<span><span class="co">#&gt; [133] DBI_1.2.3                numDeriv_2016.8-1.1      fastmap_1.2.0           </span></span>
<span><span class="co">#&gt; [136] rmarkdown_2.27           scales_1.3.0             grid_4.4.1              </span></span>
<span><span class="co">#&gt; [139] shinydashboard_0.7.2     Rsamtools_2.21.0         sass_0.4.9              </span></span>
<span><span class="co">#&gt; [142] patchwork_1.2.0          coda_0.19-4.1            farver_2.1.2            </span></span>
<span><span class="co">#&gt; [145] tidygraph_1.3.1          scatterpie_0.2.3         mgcv_1.9-1              </span></span>
<span><span class="co">#&gt; [148] yaml_2.3.8               AnnotationForge_1.47.1   rtracklayer_1.65.0      </span></span>
<span><span class="co">#&gt; [151] cli_3.6.3                purrr_1.0.2              writexl_1.5.0           </span></span>
<span><span class="co">#&gt; [154] txdbmaker_1.1.1          webshot_0.5.5            lifecycle_1.0.4         </span></span>
<span><span class="co">#&gt; [157] mvtnorm_1.2-5            sessioninfo_1.2.2        rintrojs_0.3.4          </span></span>
<span><span class="co">#&gt; [160] BiocParallel_1.39.0      annotate_1.83.0          gtable_0.3.5            </span></span>
<span><span class="co">#&gt; [163] rjson_0.2.21             ggridges_0.5.6           parallel_4.4.1          </span></span>
<span><span class="co">#&gt; [166] ape_5.8                  limma_3.61.2             jsonlite_1.8.8          </span></span>
<span><span class="co">#&gt; [169] colourpicker_1.3.0       seriation_1.5.5          bitops_1.0-7            </span></span>
<span><span class="co">#&gt; [172] ggplot2_3.5.1            bit64_4.0.5              assertthat_0.2.1        </span></span>
<span><span class="co">#&gt; [175] yulab.utils_0.1.4        heatmaply_1.5.0          geneLenDataBase_1.41.2  </span></span>
<span><span class="co">#&gt; [178] bs4Dash_2.3.3            bdsmatrix_1.3-7          jquerylib_0.1.4         </span></span>
<span><span class="co">#&gt; [181] highr_0.11               GOSemSim_2.31.0          lazyeval_0.2.2          </span></span>
<span><span class="co">#&gt; [184] shiny_1.8.1.1            dynamicTreeCut_1.63-1    htmltools_0.5.8.1       </span></span>
<span><span class="co">#&gt; [187] enrichplot_1.25.0        rappdirs_0.3.3           glue_1.7.0              </span></span>
<span><span class="co">#&gt; [190] httr2_1.0.1              XVector_0.45.0           RCurl_1.98-1.14         </span></span>
<span><span class="co">#&gt; [193] treeio_1.29.0            ComplexUpset_1.3.3       gridExtra_2.3           </span></span>
<span><span class="co">#&gt; [196] igraph_2.0.3             R6_2.5.1                 tidyr_1.3.1             </span></span>
<span><span class="co">#&gt; [199] gplots_3.1.3.1           fdrtool_1.2.17           labeling_0.4.3          </span></span>
<span><span class="co">#&gt; [202] GenomicFeatures_1.57.0   cluster_2.1.6            rngtools_1.5.2          </span></span>
<span><span class="co">#&gt; [205] bbmle_1.0.25.1           aplot_0.2.3              DelayedArray_0.31.3     </span></span>
<span><span class="co">#&gt; [208] tidyselect_1.2.1         GOstats_2.71.0           ggforce_0.4.2           </span></span>
<span><span class="co">#&gt; [211] xml2_1.3.6               munsell_0.5.1            KernSmooth_2.23-24      </span></span>
<span><span class="co">#&gt; [214] goseq_1.57.2             data.table_1.15.4        htmlwidgets_1.6.4       </span></span>
<span><span class="co">#&gt; [217] fgsea_1.31.0             ComplexHeatmap_2.21.0    RColorBrewer_1.1-3      </span></span>
<span><span class="co">#&gt; [220] rlang_1.1.4              rentrez_1.2.3            lpsymphony_1.33.0       </span></span>
<span><span class="co">#&gt; [223] fansi_1.0.6</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by a1 a2, a1 a2, a1 a2, <a href="https://federicomarini.github.io" class="external-link">Federico Marini</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
